feline/src/main/java/com/spotify/feline/FelineRuntime.java

```diff
@@ -17,33 +17,66 @@
 package com.spotify.feline;
 
 import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.function.Consumer;
 
 public class FelineRuntime {
   public static final ThreadLocal<Boolean> STATE = ThreadLocal.withInitial(() -> false);
 
-  private static final List<Consumer<String>> blockingCallConsumers = new CopyOnWriteArrayList<>();
+  private static final List<Consumer<Map<String, Object>>> blockingCallConsumers =
+      new CopyOnWriteArrayList<>();
 
-  public static void addConsumerFirst(final Consumer<String> blockingCallConsumer) {
+  // Needed for removals of legacy method
+  private static final Map<Consumer<String>, Consumer<Map<String, Object>>> legacyMapping =
+      new ConcurrentHashMap<>();
+
+  public static void addMapConsumerFirst(final Consumer<Map<String, Object>> blockingCallConsumer) {
     blockingCallConsumers.add(0, blockingCallConsumer);
   }
 
-  public static void addConsumerLast(final Consumer<String> blockingCallConsumer) {
+  public static void addConsumerFirst(final Consumer<String> blockingCallConsumer) {
```

Comment:
Should we already now mark this as deprecated?

Reply:
Since we decided that both onEnter and onExit have valid use cases, it's probably best to not deprecate anything

Reply:
Agreed